backend/Match/views.py

```diff
@@ -5,37 +5,35 @@
 from rest_framework import status
 from django.shortcuts import redirect
 
-from User.models import User
-from .models import Match
 from Tokens.models import MatchToken
-from Tokens.serializers import MatchTokenSerializer
 from shared_utilities.GameSetupManager import GameSetupManager
 from django.utils.decorators import method_decorator
-from shared_utilities.decorators import must_be_authenticated, \
-											valid_serializer_in_body
+from shared_utilities.decorators import must_be_authenticated
 
+class MatchView(APIView):
+    @method_decorator(must_be_authenticated)
+    def get(self, request):
+        token_str = request.query_params.get('token')
+        if not token_str:
+            return Response({'error': 'Token parameter is required'}, status=status.HTTP_400_BAD_REQUEST)
 
-class LaunchSingleMatchView(APIView):
-	@method_decorator(must_be_authenticated)
-	@method_decorator(valid_serializer_in_body(MatchTokenSerializer))
-	def post(self, request):
-		try:
-			token = MatchToken.objects.get(token=request.data.get('token'))
-		except MatchToken.DoesNotExist:
-			return Response({'error': 'Invalid token'}, status=status.HTTP_400_BAD_REQUEST)
-
-		if token.is_active == False or token.is_expired():
-			return Response({'error': 'Expired token'}, status=status.HTTP_403_FORBIDDEN)
-		if token.user_left_side.id != request.user.id:
-			return Response({'error': 'You are not the host user in the token'}, status=status.HTTP_403_FORBIDDEN)
-		
-		match, host_player, guest_player = GameSetupManager.create_match_and_its_players(token)
+        try:
+            token = MatchToken.objects.get(token=token_str)
+        except MatchToken.DoesNotExist:
+            return Response({'error': 'Invalid token'}, status=status.HTTP_400_BAD_REQUEST)
 
-		if not all([match, host_player, guest_player]):
-			return Response({'error': 'Something went wrong when creating the match and players'}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
-		
-		pong_match_url = f'http://localhost:80/pong/{match.id}?token={token.token}'
-		return redirect(pong_match_url) # this is prob not correct as this is a single-page app...
+        if not token.is_active or token.is_expired():
+            return Response({'error': 'Expired token'}, status=status.HTTP_403_FORBIDDEN)
+        if token.user_left_side.id != request.user.id:
+            return Response({'error': 'You are not the host user in the token'}, status=status.HTTP_403_FORBIDDEN)
+
+        match, player_left, player_right = GameSetupManager.create_match_and_its_players(token)
+
+        if not all([match, player_left, player_right]):
+            return Response({'error': 'Something went wrong when creating the match and players'}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+        pong_match_url = f'http://localhost:80/pong/{match.id}?token={token.token}'
```

Comment:
the url localhost etc part could be moved into a constant too, but maybe on the next branch? this would be annoying to find if / when we change the port / domain etc.

Reply:
sure, sounds good.